frontend/lib/resolution/resolution-queries.cpp

```diff
@@ -2846,27 +2841,52 @@ static bool resolveFnCallSpecial(Context* context,
   return false;
 }
 
-static CallResolutionResult
-resolveFnCallDomain(Context* context,
-                    const Call* call,
-                    const CallInfo& ci,
-                    const Scope* inScope,
-                    const PoiScope* inPoiScope) {
-  // TODO: a compiler-generated type constructor would be simpler, but we
-  // don't support default values on compiler-generated methods because the
-  // default values require existing AST.
+static bool isKeywordTypeCtor(Context* context,
```

Comment:
I think the name here is misleading because it also handles resolving non-keyword types via `resolveBuiltinTypeCtor`: `*` and `?`. You could rename _this one_ to `resolveFnCallSpecialType`?